controllers/servingruntime_controller.go

```diff
@@ -437,32 +429,31 @@ func (r *ServingRuntimeReconciler) getRuntimesSupportingPredictor(ctx context.Co
 	}
 
 	restProxyEnabled := r.ConfigProvider.GetConfig().RESTProxy.Enabled
-	srnns := make([]types.NamespacedName, 0, len(runtimes.Items))
+	srNameMap := make(map[string]bool)
+	for i := range csrs.Items {
+		crt := &csrs.Items[i]
+		if crt.Spec.IsMultiModelRuntime() && runtimeSupportsPredictor(&crt.Spec, p, restProxyEnabled, crt.Name) {
```

Comment:
Maybe move this check from these two loops to the single loop below? Then it only needs to be in one place